istioctl/pkg/precheck/precheck.go

```diff
@@ -122,10 +118,126 @@ See %s for more information about causes and resolutions.`, url.ConfigAnalysis)
 		fmt.Sprintf("The severity level of precheck at which to display messages. Valid values: %v", diag.GetAllLevelStrings()))
 	cmd.PersistentFlags().StringVarP(&msgOutputFormat, "output", "o", formatting.LogFormat,
 		fmt.Sprintf("Output format: one of %v", formatting.MsgOutputFormatKeys))
+	cmd.PersistentFlags().StringVarP(&fromCompatibilityVersion, "from-version", "f", "",
+		"check changes since the provided version")
 	opts.AttachControlPlaneFlags(cmd)
 	return cmd
 }
 
+func checkFromVersion(ctx cli.Context, revision, version string) (diag.Messages, error) {
+	cli, err := ctx.CLIClientWithRevision(revision)
+	if err != nil {
+		return nil, err
+	}
+	major, minors, ok := strings.Cut(version, ".")
+	if !ok {
+		return nil, fmt.Errorf("invalid version %v, expected format like '1.0'", version)
+	}
+	if major != "1" {
+		return nil, fmt.Errorf("expected major version 1, got %v", version)
+	}
+	minor, err := strconv.Atoi(minors)
+	if err != nil {
+		return nil, fmt.Errorf("minor version is not a number: %v", minors)
+	}
+
+	var messages diag.Messages = make([]diag.Message, 0)
+	if minor <= 20 {
+		// VERIFY_CERTIFICATE_AT_CLIENT and ENABLE_AUTO_SNI
```

Comment:
This doesn't feel scalable to me from a process/maintainability perspective. Are we planning to put all of the permutations of changed env vars in different versions here in the code? Have we considered serializing this to a YAML file or something?

Reply:
The intent here is we have actual code that checks if they are impacted ,so we don't just blast 100% of users "these N things changed". Only probably ~2% of users are impacted by VERIFY_CERTIFICATE_AT_CLIENT for example -- this filters them out.

Reply:
Ack, agreed. I'm just wondering if there's a way to generalize "things that changed in this version". We still have to implement the checks in code, but can we serialize the version and the feature in a YAML file similar to what exists in the helm chart? I'm just trying to see how this doesn't turn into a forest of if statements as Istio has more releases

Reply:
Hmm, I was thinking we would prune them after N-2 versions so it would be bounded. I suppose we could support even more in `istioctl` than we do in compatibility versions if we want, though.

Reply:
Ah that's true as long as we explicitly only support N-2 compatibility versions. That's less of a concern for me then; I still think a machine readable format will be useful so we're not duplicating this info in 3 places (pre check, helm chart, and public docs). But won't block on it; it can always be an optimization